include/aos/sm/networkmanager.hpp

```diff
@@ -360,6 +360,60 @@ class TrafficMonitorItf {
         = 0;
 };
 
+/**
+ * Namespace provider interface.
+ */
+class NamespaceProviderItf {
+public:
+    /**
+     * Destructor.
+     */
+    virtual ~NamespaceProviderItf() = default;
+
+    /**
+     * Creates network namespace.
+     * @param instanceID instance ID.
+     * @return Error.
+     */
+    virtual Error CreateNetworkNamespace(const String& instanceID) const = 0;
+
+    /**
+     * Returns network namespace path.
+     *
+     * @param instanceID instance ID.
+     * @param[out] netnsPath network namespace path.
+     * @return Error.
+     */
+    virtual Error GetNetworkNamespacePath(const String& instanceID, String& netnsPath) const = 0;
```

Comment:
Can we have 'RetwithError<StaticString<cFilePathLen>>'? We use it for string value a lot as I see.